core/src/main/java/org/wso2/carbon/kernel/internal/DataHolder.java

```diff
@@ -81,4 +84,22 @@ public void setRuntimeManager(RuntimeManager runtimeManager) {
     public void setOptConfigResolver(Optional<ConfigResolver> optConfigResolver) {
         this.optConfigResolver = optConfigResolver;
     }
+
+    /**
+     * Getter method of ${@link ConfigProvider}.
+     *
+     * @return optConfigProvider
+     */
+    public Optional<ConfigProvider> getOptConfigProvider() {
+        return optConfigProvider;
+    }
+
+    /**
+     * Setter method of ${@link ConfigProvider}.
+     *
+     * @param optConfigProvider
+     */
+    public void setOptConfigProvider(Optional<ConfigProvider> optConfigProvider) {
```

Comment:
Why do we have to get and set a optional value directly. IMO, we can use the Optional return type with the getter method but for setter, you can directly use the value. Or do we have a proper scenario here to use optional?